ideascube/monitoring/forms.py

```diff
@@ -143,14 +155,13 @@ class ReturnForm(forms.Form):
 
     def clean_loan(self):
         barcode = self.cleaned_data['loan']
-        try:
-            loan = Loan.objects.due().get(specimen__barcode=barcode)
-        except Loan.DoesNotExist:
-            msg = _('Item with barcode {barcode} is not loaned.')
+        loan = Loan.objects.due().filter(specimen__barcode=barcode).first()
+        if loan is None:
+            loan = Loan.objects.due().filter(specimen__serial=barcode).first()
+        if loan is None:
+            msg = _('Item with barcode or serial number {barcode} is not '
+                    'loaned.')
             raise forms.ValidationError(msg.format(barcode=barcode))
-        except Loan.MultipleObjectsReturned:
-            # Should araise only after migration 0.3.2 to 0.3.0.
-            loan = Loan.objects.due().filter(specimen__barcode=barcode).first()
```

Comment:
This code is not removed.
As `filter(...).first()` "includes" `get(...)`, I always use the more global version.